File: MoodByte_Interfaz/Modelo/Usuario.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Modelo
{
    public class Usuario
    {
        public long Id { get; set; }
        public string NombreUsuario { get; set; }
        public string NombreCompleto { get; set; }
        public int Edad { get; set; }
        public int Nivel { get; set; }
        public double ExpAcumulada { get; set; }
        public string Password { get; set; }
        public DateOnly FechaRegistro { get; set; }
        public DateOnly FechaNacimiento { get; set; }
        [JsonConverter(typeof(JsonStringEnumConverter))]
        public Genero Genero { get; set; }
        [JsonConverter(typeof(JsonStringEnumConverter))]
        public TipoUsuario TipoUsuario { get; set; }
    }
}

File: MoodByte_App/app/src/main/java/com/example/moodbyte/domain/model/Usuario.kt
package com.example.moodbyte.domain.model

import com.example.moodbyte.data.local.entities.UsuarioEntity
import com.example.moodbyte.data.remote.dtos.UsuarioDto
import java.time.LocalDate
import kotlin.math.exp

data class Usuario(
    val id: Long? = null,
    var nombreCompleto: String,
    var nombreUsuario: String,
    var password:String,
    val edad: Int,
    val genero: Genero,
    val tipoUsuario: TipoUsuario,
    val fechaRegistro: LocalDate,
    val fechaNacimiento: LocalDate,
    val nivel: Int,
    val expAcumulada: Double
)

fun Usuario.toDto()= UsuarioDto(
    id=id,
    nombreCompleto=nombreCompleto,
    nombreUsuario=nombreUsuario,
    password=password,
    edad=edad,
    genero=genero.toString(),
    tipoUsuario=tipoUsuario.toString(),
    fechaRegistro=fechaRegistro.toString(),
    fechaNacimiento=fechaNacimiento.toString(),
    nivel=nivel,
    expAcumulada= expAcumulada
)

fun Usuario.toEntity()= UsuarioEntity(
    id = id,
    nombreCompleto = nombreCompleto,
    nombreUsuario = nombreUsuario,
    password = password,
    edad = edad,
    genero = genero.toString(),
    tipoUsuario = tipoUsuario.toString(),
    fechaRegistro = fechaRegistro.toString(),
    fechaNacimiento = fechaNacimiento.toString(),
    nivel = nivel,
    expAcumulada = expAcumulada
)

